File: go/dreamdojo.go
package aegis

import (
	"context"

	"github.com/ai-aegis-yatav/aegis-sdk/go/internal"
)

// DreamdojoService — V3 DreamDojo embodied-agent validation.
type DreamdojoService struct {
	t *internal.Transport
}

func (s *DreamdojoService) ValidateAction(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/dreamdojo/validate-action", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *DreamdojoService) ValidateInput(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/dreamdojo/validate-input", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *DreamdojoService) ValidatePipeline(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/dreamdojo/validate-pipeline", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *DreamdojoService) ValidateLatent(ctx context.Context, body map[string]any) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "POST", "/v3/dreamdojo/validate-latent", body, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

func (s *DreamdojoService) Embodiments(ctx context.Context) (map[string]any, error) {
	var resp map[string]any
	if err := s.t.Do(ctx, "GET", "/v3/dreamdojo/embodiments", nil, &resp); err != nil {
		return nil, err
	}
	return resp, nil
}

File: rust/src/resources/dreamdojo.rs
use std::sync::Arc;

use reqwest::Method;
use serde_json::Value;

use crate::error::Result;
use crate::transport::Transport;

pub struct Dreamdojo {
    transport: Arc<Transport>,
}

impl Dreamdojo {
    pub(crate) fn new(transport: Arc<Transport>) -> Self {
        Self { transport }
    }

    pub async fn validate_action(&self, body: &Value) -> Result<Value> {
        self.transport
            .request(Method::POST, "/v3/dreamdojo/validate-action", Some(body))
            .await
    }

    pub async fn validate_input(&self, body: &Value) -> Result<Value> {
        self.transport
            .request(Method::POST, "/v3/dreamdojo/validate-input", Some(body))
            .await
    }

    pub async fn validate_pipeline(&self, body: &Value) -> Result<Value> {
        self.transport
            .request(Method::POST, "/v3/dreamdojo/validate-pipeline", Some(body))
            .await
    }

    pub async fn validate_latent(&self, body: &Value) -> Result<Value> {
        self.transport
            .request(Method::POST, "/v3/dreamdojo/validate-latent", Some(body))
            .await
    }

    pub async fn embodiments(&self) -> Result<Value> {
        self.transport
            .request(Method::GET, "/v3/dreamdojo/embodiments", None::<&()>)
            .await
    }
}
